tests/integration/cass_integration_test.go

```diff
@@ -0,0 +1,168 @@
+func clusterInstallPreconditions(t *testing.T, helmOptions *helm.Options, releaseName string) {
+
+	assert.NotNil(t, helmOptions.KubectlOptions)
+	assert.NotNil(t, helmOptions)
+
+	util.CreateNamespace(t, helmOptions)
+	clusterIdentity := util.CreateClusterIdentity(t, helmOptions.KubectlOptions, clusterName, releaseName)
+
+	cleanupCluster(t, helmOptions, releaseName, clusterIdentity)
+
+	repoAdd(t, helmOptions, "k8ssandra", "https://helm.k8ssandra.io/")
+	repoUpdate(t, helmOptions)
+}
+
+// operatorInstallPreconditions provides test cleanup and preconditions prior to test function execution
+func operatorInstallPreconditions(t *testing.T, helmOptions *helm.Options, releaseName string) {
+
+	assert.NotNil(t, helmOptions.KubectlOptions)
+	assert.NotNil(t, helmOptions)
+
+	util.CreateNamespace(t, helmOptions)
+
+	cleanupOperator(t, helmOptions, releaseName, util.CreateOperatorIdentity())
+
+	repoAdd(t, helmOptions, "k8ssandra", "https://helm.k8ssandra.io/")
+	repoUpdate(t, helmOptions)
+}
+
+func repoAdd(t *testing.T, helmOptions *helm.Options, name string, url string) bool {
+
+	var addErr error
+	addResult, addErr := helm.RunHelmCommandAndGetOutputE(t, helmOptions, "repo", "add", name, url)
+	util.Log(t, "Repo", "Add", addResult, addErr)
+	return (addErr == nil)
+}
+
+func repoUpdate(t *testing.T, helmOptions *helm.Options) bool {
+
+	var updateErr error
+	updateResult, updateErr := helm.RunHelmCommandAndGetOutputE(t, helmOptions, "repo", "update")
+	util.Log(t, "Repo", "Update", updateResult, updateErr)
+	return (updateErr == nil)
+}
+
+func uninstall(t *testing.T, helmOptions *helm.Options, releaseName string) bool {
+
+	var uninstallErr error
+	uninstallResult, uninstallErr := helm.RunHelmCommandAndGetOutputE(t, helmOptions, "uninstall", releaseName, "--no-hooks")
+	util.Log(t, "Helm", "Uninstall", uninstallResult, uninstallErr)
+	return (uninstallErr == nil)
+}
+
+func install(t *testing.T, helmOptions *helm.Options, releaseName string, chartName string) bool {
+
+	var installErr error
+	installResult, installErr := helm.RunHelmCommandAndGetOutputE(t, helmOptions, "install", "-n", k8ssandraNamespace,
+		releaseName, chartName, "--create-namespace", "--insecure-skip-tls-verify")
+
+	util.Log(t, "Helm", "Install", installResult, installErr)
+	return (installErr == nil)
+}
+
+// cleanupOperator cleanup of operator specific resources.
+func cleanupOperator(t *testing.T, helmOptions *helm.Options, releaseName string, identity util.CRDIdentity) {
+
+	if util.IsNamespaceExisting(t, helmOptions.KubectlOptions, k8ssandraNamespace) {
+
+		kubeOptions := k8s.NewKubectlOptions("", "", k8ssandraNamespace)
+		helmOptions := &helm.Options{KubectlOptions: kubeOptions}
+
+		util.Annotate(t, helmOptions, releaseName)
+		util.DeleteDeployment(t, helmOptions, chartName)
+		util.DeleteOperatorCRD(t, kubeOptions, identity, deleteCRDTimeoutSeconds)
+		util.DeleteRelease(t, helmOptions, releaseName)
+		util.DeletePodsByNamespace(t, helmOptions)
```

Comment:
Pods are almost always owned by some other object, e.g., Deployment, StatefulSet, etc. Deletes are typically cascaded by the owning object.